File: java/src/main/java/com/workshop/architecture/fitness/layered/infrastructure/external_invoice_provider/ExternalInvoiceProviderUpsertRequest.java
package com.workshop.architecture.fitness.layered.infrastructure.external_invoice_provider;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.Map;

public record ExternalInvoiceProviderUpsertRequest(
        @NotBlank String customerReference,
        @NotBlank String contractReference,
        @Min(0) int amountInCents,
        @NotBlank String currency,
        @NotNull LocalDate dueDate,
        @NotNull ExternalInvoiceProviderStatus status,
        String description,
        String externalCorrelationId,
        Map<String, String> metadata
) {
}

File: python/src/workshop_api/fitness/shared/external_invoice_provider_upsert_request.py
from datetime import date

from pydantic import BaseModel, ConfigDict, Field

from workshop_api.fitness.shared.invoice_provider_status import ExternalInvoiceProviderStatus


class ExternalInvoiceProviderUpsertRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    customer_reference: str = Field(alias="customerReference")
    contract_reference: str = Field(alias="contractReference")
    amount_in_cents: int = Field(alias="amountInCents")
    currency: str
    due_date: date = Field(alias="dueDate")
    status: ExternalInvoiceProviderStatus
    description: str | None = None
    external_correlation_id: str | None = Field(default=None, alias="externalCorrelationId")
    metadata: dict[str, str] | None = None
